usaspending_api/awards/views.py

```diff
@@ -121,6 +119,9 @@ def get_queryset(self):
         if not (fiscal_year and awarding_agency_id):
             raise InvalidParameterException('Missing one or more required query parameters: fiscal_year, awarding_agency_id')
 
+        if not check_valid_toptier_agency(awarding_agency_id):
```

Comment:
same error as above.  need to check if awarding_agency_id actually corresponds to an agency b4 calling .first() in check_valid_toptier_agency